### PR TITLE
Replace DeepCopyTime with metav1.Time

### PR DESCRIPTION
**Pull request by tfussell**
## Checklist

- [ ] Consider SIG UX feedback.
- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by tfussell** [comment]
> LGTM, I hope this doesn't break compatibility too much.

Same here. I have trouble telling what the effect will be. I will see if I can test it somehow.

File: pkg/apis/infrastructure/v1alpha2/common_cluster_status_funcs.go [hunk 1]
```diff
@@ -286,14 +288,14 @@ func withCondition(conditions []CommonClusterStatusCondition, condition CommonCl
 		for _, g := range conditionGroups {
 			if len(p) == 0 {
 				p = append(p, g...)
-				conditionGroups = conditionGroups[1:len(conditionGroups)]
+				conditionGroups = conditionGroups[1:]
```
Comment (tfussell) [inline pkg/apis/infrastructure/v1alpha2/common_cluster_status_funcs.go:291]: Some extra cleanup--semantics are identical as far as I know.

**Comment by kopiczko** [comment]
> > LGTM, I hope this doesn't break compatibility too much.
> 
> Same here. I have trouble telling what the effect will be. I will see if I can test it somehow.

Maybe try vendoring in aws-operator.

**Comment by tfussell** [comment]
I tried this in `aws-operator@master` and didn't have any issues. In `aws-operator@legacy`, I discovered that `statusresource` had to be updated, but there isn't much to it. I'll do that tomorrow. Going with this for now.

**Pull request merged**